daemon/daemon.go

```diff
@@ -468,7 +468,19 @@ func (d *Daemon) Start() {
 func (d *Daemon) Stop() error {
 	d.tomb.Kill(nil)
 	d.snapdListener.Close()
+
 	if d.snapListener != nil {
+		// stop running hooks first
+		// and do it more gracefully if we are restarting
+		hookMgr := d.overlord.HookManager()
+		if d.overlord.State().Restarting() {
+			// wait max 10 minutes, same as hooks timeout
+			const maxWaitIter = 10 * 60
```

Comment:
Can we bind this constant to the timeout we have in hookmgr so that they never get out of sync? Perhaps this entire waiting loop could be moved to `hookmgr.WaitForRunningHooks()` or some such?

Reply:
let me see what I can do